pkg/object/objects.go

```diff
@@ -385,28 +385,87 @@ func ManifestDiff(a, b string, verbose bool) (string, error) {
 
 // ManifestDiffWithSelect checks the manifest differences with selected and ignored resources.
 // The selected filter will apply before the ignored filter.
-func ManifestDiffWithSelectAndIgnore(a, b, selectResources, ignoreResources string, verbose bool) (string, error) {
+func ManifestDiffWithRenameSelectIgnore(a, b, renameResources, selectResources, ignoreResources string, verbose bool) (string, error) {
+	rnm := getKeyValueMap(renameResources)
 	sm := getObjPathMap(selectResources)
 	im := getObjPathMap(ignoreResources)
-	aosm, err := filterResourceWithSelectAndIgnore(a, sm, im)
+
+	ao, err := ParseK8sObjectsFromYAMLManifest(a)
 	if err != nil {
 		return "", err
 	}
-	bosm, err := filterResourceWithSelectAndIgnore(b, sm, im)
+	aom := ao.ToMap()
+
+	bo, err := ParseK8sObjectsFromYAMLManifest(b)
+	if err != nil {
+		return "", err
+	}
+	bom := bo.ToMap()
+
+	if len(rnm) != 0 {
+		aom, err = renameResource(aom, rnm)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	aosm, err := filterResourceWithSelectAndIgnore(aom, sm, im)
+	if err != nil {
+		return "", err
+	}
+	bosm, err := filterResourceWithSelectAndIgnore(bom, sm, im)
 	if err != nil {
 		return "", err
 	}
 
 	return manifestDiff(aosm, bosm, im, verbose)
 }
 
-// filterResourceWithSelectAndIgnore filter the input resources with selected and ignored filter.
-func filterResourceWithSelectAndIgnore(a string, sm, im map[string]string) (map[string]*K8sObject, error) {
-	ao, err := ParseK8sObjectsFromYAMLManifest(a)
-	if err != nil {
-		return nil, err
+// renameResource filter the input resources with selected and ignored filter.
+func renameResource(iom map[string]*K8sObject, rnm map[string]string) (map[string]*K8sObject, error) {
+	oom := make(map[string]*K8sObject)
+	for name, obj := range iom {
+		isRenamed := false
+		for fromPat, toPat := range rnm {
+			fromRe, err := buildResourceRegexp(strings.TrimSpace(fromPat))
+			if err != nil {
+				return nil, fmt.Errorf("error building the resource regexp: %v", err)
+			}
+			if fromRe.MatchString(name) {
+				fromList := strings.Split(name, ":")
```

Comment:
shouldn't this be `fromList := strings.Split(fromPat, ":")`?

Reply:
The following log message must be misleading. 
Clarification: the fromPat is intended to only be used for matching. Once matching is done, we use the old name to fill up missing parts just in case that newPat a wildcard.

Reply:
I updated the log messages and comments. also added tests to cover wildcard cases.